devicemodel/hw/vga.c

```diff
@@ -339,7 +338,8 @@ vga_mem_rd_handler(struct vmctx *ctx, uint64_t addr, void *arg1)
 		/*
 		 * monochrome text mode: base 0xb0000 size 32kb
 		 */
-		assert(0);
+		pr_err("VGA: doesn't support monochrome text mode.\n");
+		return 0;
```

Comment:
Please make sure the return value is aligned with spec. And the data should be aligned between pio and mmio reading.